File: src/braunchem/utils/filters/containers.py
"""
Fenced divs
"""

import panflute as pf


def latex(text):
    return pf.RawBlock(text, format="latex")


def div2env(elem, doc, debug=False):
    if isinstance(elem, pf.Div) and doc.format == "latex":
        env = elem.classes[0]

        begin = latex(f"\n\\begin{{{env}}}\n")
        end = latex(f"\n\\end{{{env}}}\n")

        return pf.Div(begin, elem, end)

    return


def main(doc=None):
    return pf.run_filter(div2env, doc=doc)


if __name__ == "__main__":
    main()

File: src/braunchem/utils/filters/containers.lua
if FORMAT:match 'latex' then
    local title

    local gettitle = {
        Header = function(elem)
            local text = elem.content
            if title == nil then
                title = text
                return {}
            else
                return pandoc.RawInline('latex', '\\subheader{' .. text .. '}')
            end
        end
    }

    function Div(elem)
        local class = elem.classes[1]

        title = nil
        elem.content = elem.content:walk(gettitle)

        if title == nil then
            return {
                pandoc.RawInline('latex', '\\begin{' .. class .. '}{}'),
                elem,
                pandoc.RawInline('latex', '\\end{' .. class .. '}'),
            }
        end
        return {
            pandoc.RawInline('latex', '\\begin{' .. class .. '}{' .. title .. '}'),
            elem,
            pandoc.RawInline('latex', '\\end{' .. class .. '}'),
        }
    end
end
